docs/DocumentOutlineAndTableOfContents.md

```diff
@@ -1,39 +1,80 @@
+Since `fpdf2.3.3`, you can use the [`start_section`](fpdf/fpdf.html#fpdf.fpdf.FPDF.start_section) method to add entries in the internal "outline" table, which is used to render both the outline and ToC.
 
 Note that by default, calling `start_section` only records the current position in the PDF and renders nothing.
-However, you can configure **global title styles** by calling [`set_section_title_styles`](fpdf/fpdf.html#fpdf.fpdf.FPDF.set_section_title_styles),
-after which call to `start_section` will render titles visually using the styles defined.
+However, you can configure **global title styles** by calling [`set_section_title_styles`](fpdf/fpdf.html#fpdf.fpdf.FPDF.set_section_title_styles), after which call to `start_section` will render titles visually using the styles defined.
+
+To provide a document outline to the PDF you generate, you just have to call the `start_section` method for every hierarchical section you want to define.
+
+### Nested outlines
 
-To provide a document outline to the PDF you generate, you just have to call the `start_section` method
-for every hierarchical section you want to define.
+Outlines can be nested by specifying different levels. Higher-level outlines (e.g., level 0) appear at the top, while sub-levels (e.g., level 1, level 2) are indented.
 
-If you also want to insert a table of contents somewhere,
-call [`insert_toc_placeholder`](fpdf/fpdf.html#fpdf.fpdf.FPDF.insert_toc_placeholder)
-wherever you want to put it.
-Note that a page break will always be triggered after inserting the table of contents.
+```python
+pdf.start_section(name="Chapter 1: Introduction", level=0)
+pdf.start_section(name="Section 1.1: Background", level=1)
+```
+
+---
+
+## Table of Contents
+
+Quoting [Wikipedia](https://en.wikipedia.org/wiki/Table_of_contents), a **table of contents** is:
+> a list, usually found on a page before the start of a written work, of its chapter or section titles or brief descriptions with their commencing page numbers.
 
-## With HTML ##
+### Inserting a Table of Contents
 
-When using [`FPDF.write_html`](HTML.md), a document outline is automatically built.
-You can insert a table of content with the special `<toc>` tag.
+Use the [`insert_toc_placeholder`](fpdf/fpdf.html#fpdf.fpdf.FPDF.insert_toc_placeholder) method to define a placeholder for the ToC. A page break is triggered after inserting the ToC.
+
+**Parameters:**
+- **render_toc_function**: Function called to render the ToC, receiving two parameters: `pdf`, an FPDF instance, and `outline`, a list of `fpdf.outline.OutlineSection`.
+- **pages**: The number of pages that the ToC will span, including the current one. A page break occurs for each page specified.
+- **allow_extra_pages**: If `True`, allows unlimited extra pages for the ToC. Note that enabling this may cause discrepancies with pre-rendered page numbers. Using page labels to create a separate numbering style for the ToC is recommended.
```

Comment:
To be honest, when reading _"discrepancies with pre-rendered page numbers"_ the potential problem is quite vague to me: what do you mean by _"pre-rendered page numbers"_?

Also, you may want to add a link there to the new `docs/PageLabels.md` page